### PR TITLE
Added mean and median for arrays

### PR DESCRIPTION
**Pull request by lfrancioli**

### CODE REVIEW

File: docs/reference/HailExpressionLanguage.md [hunk 1]
```diff
@@ -87,6 +87,8 @@ Several Hail commands provide the ability to perform a broad array of computatio
  - Numeric Array Operations:
      - min: `arr.min` -- valid only for numeric arrays, returns the minimum value
      - max: `arr.max` -- valid only for numeric arrays, returns the minimum value
+     - median: `arr.median` -- valid only for numeric arrays, returns the median value
+     - mean: `arr.mean` -- valid only for numeric arrays, returns the mean as a `Double`
```
Comment (cseed) [inline docs/reference/HailExpressionLanguage.md]: Awesome!  I can't believe I didn't check breeze for median.

As always, only question is how we handle missing.  These will both treat missing as 0.0.  Is that what we want?  Might be OK for mean, but seems counter-intuitive for median.  Either way, can you clarify how missing is handled in the doc?

Comment (tpoterba) [inline docs/reference/HailExpressionLanguage.md]: I'd prefer to do what we do for the stats aggregator, and filter out the nulls

Comment (cseed) [inline docs/reference/HailExpressionLanguage.md]: Yes, I agree with Tim: `a.stats().mean` should equal `a.mean`.

Comment (jbloom22) [inline docs/reference/HailExpressionLanguage.md]: Agreed.

**Comment by jbloom22** [comment]
@lfrancioli, replace `x.filter(_ != null).size` with `x.count(_ != null)`

**Comment by cseed** [comment]
Sorry, your code was correct as originally written since you were calling "register...NAFilteredCollectionMethod".  It already does the filter.  Just revert your last push and it will be good to go.  Sorry about that!

**Comment by cseed** [comment]
But you should mention missing values are ignored in the docs...

**Comment by cseed** [comment]
@jbloom22 That won't work because x has type Int (etc.)

**Comment by jbloom22** [comment]
Oops! I thought x was the sequence.

**Comment by cseed** [comment]
Sorry, it is, _ has type Int so that won't work.

**Pull request merged**